build: Clone and build sonobuoy

pjbgf opened the pull request:
The upstream project hasn't been released in almost one year, which caused it to accumulate several stdlib Golang vulnerabilities. Once the release cadence upstream improves, this can be reverted.

macedogm commented on package/Dockerfile:
```diff
@@ -4,7 +4,7 @@ FROM registry.suse.com/bci/golang:1.22 AS golang
 # Builder and xx only need to support the host architecture.
 FROM --platform=$BUILDPLATFORM rancher/mirrored-tonistiigi-xx:1.3.0 AS xx
-FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.22 AS builder
+FROM --platform=$BUILDPLATFORM golang AS builder
```
Out of curiosity, why do we need to change from BCI?

pjbgf replied:
I didn't, this was redundant. It is referring to the line `3` above:
```
FROM registry.suse.com/bci/golang:1.22 AS golang
```

The PR review not showing it makes it confusing.

macedogm replied:
Ohh, I see. Thanks for clarifying.

pjbgf replied:
However, it turns out that forces cross emulation instead of cross compilation, so I will remove it.

pjbgf committed:
build: Clone and build sonobuoy
The upstream project hasn't been released in almost one year, which
caused it to accumulate several stdlib Golang vulnerabilities.
Once the release cadence upstream improves, this can be reverted.

Signed-off-by: Paulo Gomes <pjbgf@linux.com>
